### PR TITLE
doc: Correct Doxygen information

### PR DESCRIPTION
**Pull request by mrdeep1**
Dynamically build DoxygenLayout.xml to get the latest layout for the current
version of doxygen.

Update Doxyfile.in to the Ubuntu20 version.

Signed-off-by: Jon Shallow <supjps-libcoap@jpshallow.com>

### CODE REVIEW

**Comment by obgm** [comment]
> latest correct layout Update Doxyfile.in 

Hmmm:

```
$ doxygen -v
1.9.4
```

**Comment by obgm** [comment]
Thinking about it a little more, since the build system is Ubuntu 20, it makes sense to use the doxygen version that comes with that release. I just would not call it "latest" :-)

**Comment by mrdeep1** [comment]
Hopefully corrected the commit message to reflect the current situation.

File: crypto.h [hunk 1]
```diff
@@ -205,11 +205,15 @@ typedef struct {
...
+ * \param label    The label.
+ * \param labellen Length of \p label.
+ * \param random1 The random pt 1. 
+ * \param random1len Length of \p random1.
+ * \param random2 The random pt 2. 
+ * \param random2len Length of \p random2.
  * \param buf     Output buffer where the result is XORed into
- *                The buffe must be capable to hold at least
- *                \p buflen bytes.
+ * \param buflen  THe available space for \p buf
```
Comment (obgm) [inline crypto.h:215]: "THe" → "The"

**Review by obgm — COMMENTED** [review]
One small typo left.

File: crypto.h (hunk 1 above)
Comment (mrdeep1) [reply crypto.h]: Corrected and change pushed.

**Comment by obgm** [comment]
LGTM

**Pull request merged**